tpu-client-next/src/connection_workers_scheduler.rs

```diff
@@ -132,6 +135,12 @@ impl From<StakeIdentity> for QuicClientCertificate {
     }
 }
 
+impl StakeIdentity {
```

Comment:
I need to convert `&StakeIdentity` (because this is what returned from watch channel) into  `&QuicClientCertificate`. Since `QuicClientCertificate` is defined in another crate, I cannot implement `From`.

Reply:
You sure it does not work?
Orphan rules for Rust seems to be saying it should be fine.
https://doc.rust-lang.org/reference/items/implementations.html#orphan-rules

Specifically, the section on "Fundamental types", says:
> Any time a type `T` is considered [local](https://doc.rust-lang.org/reference/glossary.html#local-type), `&T`, `&mut T`, `Box<T>`, and `Pin<T>` are also considered local.
https://doc.rust-lang.org/reference/glossary.html#fundamental-type-constructors

So, if you can write
```rust
impl From<StakeIdentity> for QuicClientCertificate {
```
you should be able to write
```rust
impl From<&StakeIdentity> for &QuicClientCertificate {
```

I'm not saying you should, though.
Just saying you could if you want.

Reply:
actually you are right, I can